### PR TITLE
Improve MsgPack input/output format

### PR DESCRIPTION
**Pull request by Avogar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix possible LogicalError, support more types in MsgPack input/output format.

Detailed description / Documentation draft:
Fix possible LogicalError while inserting data with wrong type in MsgPack format by adding more type checks, support MsgPack types `bool`, `bin` and `map`, allow to insert data into Nullable columns, allow to insert/output data in/from LowCardinality columns.

### CODE REVIEW

**Comment by Avogar** [comment]
Will add more tests and will update docs

File: src/Processors/Formats/Impl/MsgPackRowInputFormat.cpp [hunk 1]
```diff
@@ -50,95 +52,213 @@ void MsgPackVisitor::reset()
...
+    auto insert_func = [&](IColumn & column_, DataTypePtr type_)
+    {
+        insertString(column_, type_, value, size);
+    };
+
+    if (checkAndInsertNullable(column, type, insert_func) || checkAndInsertLowCardinality(column, type, insert_func))
+        return;
+
+    if (!isStringOrFixedString(type))
+        throw Exception(ErrorCodes::ILLEGAL_COLUMN, "Cannot insert MessagePack string into column with type {}.", type->getName());
```
Comment (kssenii) [inline src/Processors/Formats/Impl/MsgPackRowInputFormat.cpp:161]: Should there be the same check for nested type inside nullable and lc?

**Review by kssenii — APPROVED** [review]

**Comment by Avogar** [comment]
@Mergifyio update

**Comment by Avogar** [comment]
@Mergifyio update

**Pull request merged**